WordPress/gutenberg
# Mobile - Slash inserter

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2620



## Description

This PR adds the `/` (slash) key shortcut to add a block on mobile.

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/3250
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/16064
`WordsPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/14245

**Pending issues**:
- [x] Update UI styles:
  - Add a background color to the highlighted item: `$gray-light` (light mode) and `$neutral-dim-dark` (dark mode).
  - <img src="https://user-images.githubusercontent.com/4885740/111274586-75a3f480-8635-11eb-9434-fc11c7e3f49f.png" width=300 />
- [x] ~Hide feature behind a feature flag for DEV~ once this PR is approved it'll ship to production so no DEV flag needed.
- [x] Issues with [Android and the onEnter event](https://github.com/WordPress/gutenberg/pull/29683/#issuecomment-799146552). Here's a [PR](https://github.com/WordPress/gutenberg/pull/30368) with a fix for this but it won't be a blocker to ship this feature.
- [x] React warning `React.createElement: type is invalid -- expected a string (for built-in components)`
- [x] Add E2E tests/sanity tests
- [x] Update CHANGELOG notes

## How has this been tested?
- Open the app
- Within a `Paragraph` block
- Type the `/` character following by any block name
- Expect to see the block list suggestion above the keyboard
- Once the block you typed is highlighted tap on the enter key
- Expect to see the block added

## Screenshots 

iOS Light | iOS Dark
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/117946935-08000600-b310-11eb-8e97-927e48ab7a57.gif" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/117946928-05051580-b310-11eb-9d03-4143d59acdfb.gif" width="250" /></kbd> 

Android Light | Android Dark
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/117947029-1b12d600-b310-11eb-86bb-2fa49a5b4f13.gif" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/117947014-1817e580-b310-11eb-806b-a4b1be2a9be2.gif" width="250" /></kbd> 

## Types of changes
New feature

## Checklist:
